apps/mobile/src/theme/provider/DataFetchProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { useAppSelector } from '@/src/store/hooks'
+import { selectAllChains } from '@/src/store/chains'
+import { makeSafeId } from '@/src/utils/formatters'
+import { useSafesGetOverviewForManyQuery } from '@safe-global/store/gateway/safes'
+import { SafeOverviewResult } from '@safe-global/store/gateway/types'
+import { useChainsBalance } from '@/src/hooks/useChainsBalance'
+import { selectActiveSafe } from '@/src/store/activeSafeSlice'
+
+export const DataFetchProvider = ({ children }: { children: React.ReactNode }) => {
+  const chains = useAppSelector(selectAllChains)
+  const activeSafe = useAppSelector(selectActiveSafe)
+
+  useChainsBalance(activeSafe?.address)
```

Comment:
I don't undrestand this hook. Why do we need to do useChainsBalance here? 

useChainBalance does:  const data = useSafeOverviewService(safeAddress)

useSafeOverviewService does:
const { data } = useSafesGetOverviewForManyQuery<SafeOverviewResult>(
    {
      safes,
      currency: 'usd',
      trusted: true,
      excludeSpam: true,
    },
    {
      skip: !safeAddress,
    },
  )
  
  nearly the same as the line you have below. What useChainsBalance also does is updating the SafeInfo from the call. 
  
The purpose of the different hooks is a mess. It's not obvious why we need one over the other. Something is wrong in our architeture.

Reply:
@compojoom I didn't created this hook, I just moved where it is located and renamed it better.